src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs

```diff
@@ -60,68 +60,74 @@ public ElasticsearchRequestPipelineDiagnosticListener(ElasticsearchClientInstrum
 
         public override void OnStartActivity(Activity activity, object payload)
         {
-            var uri = this.uriFetcher.Fetch(payload);
+            // By this time, samplers have already run and
+            // activity.IsAllDataRequested populated accordingly.
 
-            if (uri == null)
+            if (Sdk.SuppressInstrumentation)
             {
-                ElasticsearchInstrumentationEventSource.Log.NullPayload(nameof(ElasticsearchRequestPipelineDiagnosticListener), nameof(this.OnStartActivity));
                 return;
             }
 
-            ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
-            ActivityInstrumentationHelper.SetKindProperty(activity, ActivityKind.Client);
+            if (activity.IsAllDataRequested)
```

Comment:
Previously, we were doing some things even when `IsAllDataRequested` was false. Is it best practice now to not do anything?

Reply:
I see that other instrumentations like the HTTP one are still gathering some info even when `IsAllDataRequested` is false. But this PR is changing it so that we don't do anything. I'm concerned that by removing the basic level of information that the activity spans will be so generic you won't know what they are for.

Reply:
If `IsAllDataRequested` is set to `false`, we don't export the activity as `Processor.OnEnd` is not called. So it wouldn't really be an issue if the activity doesn't have that basic level of information as it would never be sent to `Processors`.

Reply:
Hmm... so why does the HTTP instrumentation still do some things then? https://github.com/open-telemetry/opentelemetry-dotnet/blob/caab6d63f47052d9f8fadd40485be313ce171f6e/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L85-L110

Reply:
HTTP instrumentation does not enrich the activity if `IsAllDataRequested` is set to `false`. It only propagates context as we don't want broken traces. If HTTP instrumentation does not propagate context, then the external system which receives the HTTP call will create new TraceId for the same transaction, and this will lead to broken traces. That's why, we ensure that the context is propagated regardless of the sampling decision as the other system that is receiving the HTTP call might upsample the activity and export it. In that case, we want to retain the same TraceId for that operation.

In case of Elasticsearch, there is no context propagation happening. We only enrich the activity, which wouldn't be necessary if we are not going to export that activity.

Reply:
Ok, that makes sense. Guess we would just need to de-duplicate the DisplayName setting then.